src/sysext/sysext.c

```diff
@@ -1199,6 +1200,10 @@ static int mount_overlayfs_with_op(
         if (r < 0)
                 return log_error_errno(r, "Failed to make directory '%s': %m", meta_path);
 
+        r = mac_selinux_apply(meta_path, "system_u:object_r:usr_t:s0");
```

Comment:
use mac_selinux_fix_full() to apply the label from the db. it takes two strings: one for the actual inode to relabel, and one for the path string to search in the db for that label. And these two paths can be distinct for cases like this.